nflog/nflog.go

```diff
@@ -0,0 +1,585 @@
+// Copyright 2016 Prometheus Team
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// A package implementing a garbage-collected and snapshottable append-only log of
```

Comment:
"Notification*" just seemed a bit long to drag through the entire code base, so I just picked "n" and "f" as two characters to represent it. Sure, it's not self-explaining but once you know it, it's always easy to recognize – easier than just "nlog" it seemed at least.